torch/package/package_importer.py

```diff
@@ -296,6 +299,16 @@ def _read_extern(self):
             .splitlines(keepends=False)
         )
 
+    def _read_selective_extern(self):
+        try:
+            return (
+                self.zip_reader.get_record(".data/selective_extern_packages")
+                .decode("utf-8")
+                .splitlines(keepends=False)
+            )
+        except BaseException:
+            return []
+
     def _make_module(
```

Comment:
Okay, here’s my idea. What if instead of writing a shim module out to the package, we dynamically generated it here?

So basically, if we call `_make_module_` on a selective extern node, we override the getattr for the module we make here. The basic logic for the get attr would be:
1. Check this importer's module tree. 
2. If `name` resolves to a selective intern node, call `this_importer.import_module()` and return the result
3. else, retrieve it from the "real" module.

This way we wouldn’t have to write out a shim at all; we could just the selective extern listing in the module as the sole source of truth, similar to how we do regular externs. Thoughts?